cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/app/etl/batch/ETLMapReduceTest.java

```diff
@@ -256,6 +264,90 @@ public void testS3toTPFS() throws Exception {
+
+    Method method = FileSystem.class.getDeclaredMethod("addFileSystemForTesting",
+                                                       new Class[]{URI.class, Configuration.class, FileSystem.class});
+    method.setAccessible(true);
+    method.invoke(FileSystem.class, URI.create(s3Path), conf, fs);
+
+    StreamManager streamManager = getStreamManager("myStream");
+    streamManager.createStream();
+    streamManager.send(ImmutableMap.of("header", "value"), "AAPL|10|500.32");
+
+    Schema eventSchema = Schema.recordOf("streamEvent",
+                                         Schema.Field.of("ts", Schema.of(Schema.Type.LONG)),
+                                         Schema.Field.of("headers", Schema.mapOf(Schema.of(Schema.Type.STRING),
+                                                                                 Schema.of(Schema.Type.STRING))),
+                                         Schema.Field.of("ticker", Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of("num", Schema.of(Schema.Type.INT)),
+                                         Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE)));
+
+    Schema bodySchema = Schema.recordOf("event",
+                                        Schema.Field.of("ticker", Schema.of(Schema.Type.STRING)),
+                                        Schema.Field.of("num", Schema.of(Schema.Type.INT)),
+                                        Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE)));
+
+    ETLStage source = new ETLStage("Stream", ImmutableMap.<String, String>builder()
+      .put(Properties.Stream.NAME, "myStream")
+      .put(Properties.Stream.DURATION, "10m")
+      .put(Properties.Stream.DELAY, "0d")
+      .put(Properties.Stream.FORMAT, Formats.CSV)
+      .put(Properties.Stream.SCHEMA, bodySchema.toString())
+      .put("format.setting.delimiter", "|")
+      .build());
+
+
+    ETLStage sink = new ETLStage("S3Avro",
+                                 ImmutableMap.of(Properties.S3BatchSink.SCHEMA, eventSchema.toString(),
+                                                 Properties.S3.ACCESS_KEY, "key",
+                                                 Properties.S3.ACCESS_ID, "ID",
+                                                 Properties.S3.PATH, s3Path));
+
+    ETLStage transform = new ETLStage("Projection", ImmutableMap.<String, String>of());
+
+    ETLBatchConfig batchConfig =  new ETLBatchConfig("* * * * *", source, sink, Lists.newArrayList(transform));
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, batchConfig);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "StreamToS3");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+    mrManager.start();
+    mrManager.waitForFinish(2, TimeUnit.MINUTES);
+
+    // TODO: Fix this before merging. It is broken now
```

Comment:
No it did not work. Removed the tests. Will update the JIRA with the test that was written and details